python/dgl/heterograph.py

```diff
@@ -5584,6 +5584,21 @@ def is_pinned(self):
         """
         return self._graph.is_pinned()
 
+    def record_stream(self, stream):
+        """Record the stream that is using this graph.
+
+        Parameters
+        ----------
+        stream : torch.cuda.Stream
+            The stream that is using this graph.
+
+        Returns
+        -------
+        DGLGraph
+            self.
+        """
+        return self._graph.record_stream(stream)
```

Comment:
@BarclayII semantically should this be recording the stream for all node and edge frames too?

Reply:
I would prefer recording them separately unless there is a reason to record all the streams together (e.g. some DGL's APIs require recording the graph and the features at the same time).

Reply:
Updated accordingly.